bson/bsoncodec/default_value_decoders.go

```diff
@@ -1666,11 +1666,17 @@ func (dvd DefaultValueDecoders) decodeDefault(dc DecodeContext, vr bsonrw.ValueR
 
 	eType := val.Type().Elem()
 
-	decoder, err := dc.LookupDecoder(eType)
-	if err != nil {
-		return nil, err
+	var vDecoder *ValueDecoder
+	var tDecoder *typeDecoder
+	if !(eType.Kind() == reflect.Interface && val.Len() > 0) {
```

Comment:
Delay the decoder retrieval for interface slice/array so each element will be decoded based on its type.